stage1/init/common/app.go

```diff
@@ -0,0 +1,239 @@
+}
+
+// This is separate because systemd <231 didn't support masking files,
+// only directories
+var protectKernelHiddenPaths = []string{
+	"/proc/config.gz",
+	"/proc/kallsyms",
+	"/proc/sched_debug",
+	"/proc/kcore",
+	"/proc/kmem",
+	"/proc/mem",
+}
+
+// prepareApp sets up the internal runtime context for a specific app.
+func prepareApp(p *stage1commontypes.Pod, ra *schema.RuntimeApp) (*preparedApp, error) {
+	pa := preparedApp{
+		app:             ra,
+		env:             ra.App.Environment,
+		noNewPrivileges: getAppNoNewPrivileges(ra.App.Isolators),
+	}
+	var err error
+
+	// Determine numeric uid and gid
+	u, g, err := parseUserGroup(p, ra)
+	if err != nil {
+		return nil, errwrap.Wrap(errors.New("unable to determine app's uid and gid"), err)
+	}
+	if u < 0 || g < 0 {
+		return nil, errors.New("Invalid uid or gid")
+	}
+	pa.uid = uint32(u)
+	pa.gid = uint32(g)
+
+	// Set some rkt-provided environment variables
+	pa.env.Set("AC_APP_NAME", ra.Name.String())
+	if p.MetadataServiceURL != "" {
+		pa.env.Set("AC_METADATA_URL", p.MetadataServiceURL)
+	}
+
+	// Determine capability set
+	pa.capabilities, err = getAppCapabilities(ra.App.Isolators)
+	if err != nil {
+		return nil, errwrap.Wrap(errors.New("unable to construct capabilities"), err)
+	}
+
+	// Determine mounts
+	cfd := ConvertedFromDocker(p.Images[ra.Name.String()])
+	pa.mounts, err = GenerateMounts(ra, p.Manifest.Volumes, cfd)
+	if err != nil {
+		return nil, errwrap.Wrap(errors.New("unable to compute mounts"), err)
+	}
+
+	// Compute resources
+	pa.resources, err = computeAppResources(ra.App.Isolators)
+	if err != nil {
+		return nil, errwrap.Wrap(errors.New("unable to compute resources"), err)
+	}
+
+	// Protect kernel tunables by default
+	if !p.InsecureOptions.DisablePaths {
+		pa.roPaths = append(pa.roPaths, protectKernelROPaths...)
+		pa.hiddenPaths = append(pa.hiddenDirs, protectKernelHiddenPaths...)
+		pa.hiddenDirs = append(pa.hiddenDirs, protectKernelHiddenDirs...)
+	}
+
+	// Seccomp
+	if !p.InsecureOptions.DisableSeccomp {
+		pa.seccomp, err = generateSeccompFilter(p, &pa)
+		if err != nil {
+			return nil, err
+		}
+		if pa.seccomp != nil && pa.seccomp.forceNoNewPrivileges {
+			pa.noNewPrivileges = true
+		}
+	}
+
+	// Write the systemd-sysusers config file
+	if err := generateSysusers(p, pa.app, int(pa.uid), int(pa.gid), &p.UidRange); err != nil {
+		return nil, errwrap.Wrapf("unable to generate sysusers file", err)
+	}
+
+	return &pa, nil
+}
+
+// computeAppResources processes any isolators that manipulate cgroups.
+func computeAppResources(isolators types.Isolators) (appResources, error) {
+	res := appResources{}
+	var err error
+
+	withIsolator := func(name string, f func() error) error {
+		ok, err := cgroup.IsIsolatorSupported(name)
+		if err != nil {
+			return errwrap.Wrapf("could not check for isolator "+name, err)
+		}
+
+		if !ok {
+			fmt.Fprintf(os.Stderr, "warning: resource/%s isolator set but support disabled in the kernel, skipping\n", name)
```

Comment:
ok, this is still here, and it was there also in the old code. I'll submit an issue to get rid of all `fmt.Fprintf` occurrences and replace them with respective injectable warnf loggers.